File: server/eBooks.Database/Models/Purchase.cs
using System;
using System.Collections.Generic;

namespace eBooks.Database.Models;

public partial class Purchase
{
    public int PurchaseId { get; set; }

    public int UserId { get; set; }

    public int PublisherId { get; set; }

    public int BookId { get; set; }

    public DateTime CreatedAt { get; set; }

    public decimal TotalPrice { get; set; }

    public string PaymentStatus { get; set; } = null!;

    public string PaymentMethod { get; set; } = null!;

    public string TransactionId { get; set; } = null!;

    public string? FailureMessage { get; set; }

    public string? FailureCode { get; set; }

    public string? FailureReason { get; set; }

    public virtual Book Book { get; set; } = null!;

    public virtual User Publisher { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: client/ebooks_admin/lib/models/purchases/purchase.dart
import "package:ebooks_admin/models/books/book.dart";
import "package:ebooks_admin/models/users/user.dart";
import "package:json_annotation/json_annotation.dart";

part "purchase.g.dart";

@JsonSerializable()
class Purchase {
  int? purchaseId;
  User? user;
  User? publisher;
  Book? book;
  DateTime? createdAt;
  double? totalPrice;
  String? paymentStatus;
  String? paymentMethod;
  String? transactionId;
  String? failureMessage;
  String? failureCode;
  String? failureReason;

  Purchase({
    this.purchaseId,
    this.user,
    this.publisher,
    this.book,
    this.createdAt,
    this.totalPrice,
    this.paymentMethod,
    this.paymentStatus,
    this.transactionId,
    this.failureMessage,
    this.failureCode,
    this.failureReason,
  });

  factory Purchase.fromJson(Map<String, dynamic> json) =>
      _$PurchaseFromJson(json);
  Map<String, dynamic> toJson() => _$PurchaseToJson(this);
}
